crates/ruff/src/args.rs

```diff
@@ -41,6 +41,32 @@ pub struct Args {
     pub command: Command,
     #[clap(flatten)]
     pub log_level_args: LogLevelArgs,
+    /// Either a path to a TOML configuration file (`pyproject.toml` or `ruff.toml`),
+    /// or a TOML `<KEY> = <VALUE>` pair
+    /// (such as you might find in a `ruff.toml` configuration file)
+    /// overriding a specific configuration option.
+    /// Overrides of individual settings using this option always take precedence
+    /// over all configuration files, including configuration files that were also
+    /// specified using `--config`.
+    #[arg(
+        long,
+        action = clap::ArgAction::Append,
+        value_name = "CONFIG_OPTION",
+        value_parser = ConfigArgumentParser,
+        global = true,
```

Comment:
What does `global` do? Is it necessary considering that it is defined at the top level arguments?

Reply:
It's necessary, yes -- without `global = true`, `--config` would only be accepted as a flag for the _top-level_ command (just `ruff`). With `global = true`, it's accepted for all subcommands as well as the top-level command: `ruff check`, `ruff format`, `ruff rule`, etc.

Reply:
That makes sense. Thanks for explaining